Pop-Grupa-A-Frontend/src/components/computationCockpit/AppList.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import { taskUrl } from '../../commons/ApiLinks';
+import Row from 'react-bootstrap/Row'
+import Col from 'react-bootstrap/Col'
+import Button from 'react-bootstrap/Button'
+import Table from 'react-bootstrap/Table'
+import { getAppsForUser } from '../../services/appService';
+
+class AppList extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      chosenAppId: -1,
+      userId: 1,
+      appsList: []
+    };
+  }
+
+  getApps() {
+    getAppsForUser(this.state.userId)
+      .then(response =>
+        this.setState({
+          appsList: response
+        })
+      );
+  }
+
+  componentDidMount() {
+    this.getApps();
+  }
+
+
+  onAppClick = (chosenAppId) => {
+    this.setState({ chosenAppId });
+  }
+
+  createTask = () => {
+    fetch(taskUrl, {
+      crossDomain: true,
+      method: 'post',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        app_id: this.state.chosenAppId,
+        user_id: this.state.userId
+      })
+    })
+      .then(response => response.json())
+      .then(response => {
+        this.setState({
+          chosenAppId: -1
+        })
+        alert(response.message);
+      }).then(() => this.props.tasksShouldRefresh(true))
+  }
+
+  render() {
+    var appId = this.state.chosenAppId;
```

Comment:
`const { chosenAppId, appsList } = this.state;` tak wyciągnąć ładnie ze stanu można raz.
I tam w returnie popodmieniać te wartości, żeby za każdym razem do stanu się nie odwoływać.